src/System.Windows.Extensions/src/System/Security/Cryptography/X509Certificates/X509Certificate2UI.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+
+namespace System.Security.Cryptography.X509Certificates
+{
+    public enum X509SelectionFlag
+    {
+        SingleSelection = 0x00,
+        MultiSelection = 0x01
+    }
+
+    public sealed class X509Certificate2UI
+    {
+        internal const int ERROR_SUCCESS = 0;
+        internal const int ERROR_CANCELLED = 1223;
+
+        public static void DisplayCertificate(X509Certificate2 certificate)
+        {
+            if (certificate == null)
+                throw new ArgumentNullException(nameof(certificate));
+            DisplayX509Certificate(X509Utils.DuplicateCertificateContext(certificate), IntPtr.Zero);
+        }
+
+        public static void DisplayCertificate(X509Certificate2 certificate, IntPtr hwndParent)
+        {
+            if (certificate == null)
+                throw new ArgumentNullException(nameof(certificate));
+            DisplayX509Certificate(X509Utils.DuplicateCertificateContext(certificate), hwndParent);
+        }
+
+        public static X509Certificate2Collection SelectFromCollection(X509Certificate2Collection certificates, string title, string message, X509SelectionFlag selectionFlag)
+        {
+            return SelectFromCollectionHelper(certificates, title, message, selectionFlag, IntPtr.Zero);
+        }
+
+        public static X509Certificate2Collection SelectFromCollection(X509Certificate2Collection certificates, string title, string message, X509SelectionFlag selectionFlag, IntPtr hwndParent)
+        {
+            return SelectFromCollectionHelper(certificates, title, message, selectionFlag, hwndParent);
+        }
+
+        private static void DisplayX509Certificate(SafeCertContextHandle safeCertContext, IntPtr hwndParent)
+        {
+            using (safeCertContext)
```

Comment:
This is still bad ownership semantics.  Since the only two callers to this method both have an X509Certificate2 object, I'd recommend changing the first parameter of this method to X509Certificate2, and then this line becomes

```C#
using (SafeCertContextHandle safeCertContext = X509Utils.DuplicateCertificateContext(certificate))
```

And now there's no weird "you closed my handle" problem.